services/QuillLMS/client/app/bundles/Comprehension/components/studentView/container.tsx

```diff
@@ -61,6 +63,8 @@ interface StudentViewContainerState {
   explanationSlidesCompleted: boolean;
   explanationSlideStep:  number;
   completedSteps: Array<number>;
+  imageAttributionModalOpen: boolean;
+  imageUrlToVisit: string;
```

Comment:
As far as I can tell, the container component doesn't need to know about any of this at all, and this file is already really bloated. I think we need to pull out all the image-related code into its own component and file and you can handle all of this in there.